### PR TITLE
Fix ROCM

### PR DESCRIPTION
**Pull request by wsmoses**

### CODE REVIEW

File: enzyme/CMakeLists.txt [hunk 1]
```diff
@@ -128,6 +128,11 @@ if (ENZYME_FLANG)
   add_definitions(-DFLANG=1)
 endif()
 
+option(ENZYME_ROCM "Build for non-version compliant ROCM" OFF)
```
Comment (vchuravy) [inline enzyme/CMakeLists.txt:131]: grrml

Comment (vchuravy) [reply enzyme/CMakeLists.txt:131]: Can we document which version this is? Is this ROCM 4.X or 5?

**Review by vchuravy — APPROVED** [review]

**Comment by jedbrown** [comment]
Thanks. If I've successfully managed my environment variables/build paths, then this (on CPU) fails when built with ROCm's LLVM, but not when build with mainline LLVM. And when I add `-ceed /gpu/hip`, I get that `stddef.h` error. (I don't remember if you figured out the cause of that today.)

```console
$ ./elasticity -problem FSInitial-NH-AD -E 1 -nu .3 -degree 2 -dm_plex_box_faces 1,1,1 -num_steps 1 -bc_clamp 6 -bc_traction 5 -bc_traction_5 0,0,.01 -snes_monitor -snes_converged_reason -outer_ksp_monitor_true_residual -pc_type none
```

**Comment by wsmoses** [comment]
The stddef.h error was fixed by rebuilding AMD comgr to point to the new ROCM installation (5.1.1). Specifically that involves setting the LD_LIBRARY_PATH to /home/wsmoses/rocm2/... instead of rocm.

What's the error you get when building using ROCm's LLVM? (and what are the precise CMake flags and ROCM path's)?

File: enzyme/CMakeLists.txt (hunk 1 above)
Comment (wsmoses) [reply enzyme/CMakeLists.txt:131]: 5.1.1

**Pull request merged**